<!-- Copied from world_sim.launch. Added launching slam with saved rviz file -->
<launch>
  <arg name="PathPlanner" default="false"/>
  <arg name="LabTwo" default="false"/>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="world" default="turtlebot" />
  <arg name="x_pos" default="-2.2" unless="$(eval arg('world') == 'turtlebot')"/>
  <arg name="y_pos" default="0.4" unless="$(eval arg('world') == 'turtlebot')"/>
  <arg name="x_pos" default="0.4" if="$(eval arg('world') == 'turtlebot')"/>
  <arg name="y_pos" default="2.0" if="$(eval arg('world') == 'turtlebot')"/>
  <arg name="z_pos" default="0.05"/>


  <node pkg="tf" type="static_transform_publisher" name="camera_tf" args="-1.95 -0.55 2.0 -1.58 0 -1.58 /odom /camera_link 100"/>

  <env name="GAZEBO_MODEL_PATH" value="$(find lastlab)/models:$(optenv GAZEBO_MODEL_PATH"/>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find lastlab)/worlds/$(arg world).world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <!-- Gmapping -->
  <node pkg="gmapping" type="slam_gmapping" name="turtlebot3_slam_gmapping" output="screen">
    <param name="base_frame" value="base_footprint"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_frame"  value="map"/>
    <rosparam command="load" file="$(find lastlab)/config/my_gmapping_params.yaml" />
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find lastlab)/rviz/final_config.rviz"/>

  <param name="robot_description" command="$(find xacro)/xacro $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model turtlebot3_burger -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />
  
  <group if="$(arg PathPlanner)">
    <node name="path_planner" pkg="lastlab" type="path_planner.py"/>
  </group>

  <group if="$(arg LabTwo)">
    <node name="lab2" pkg="lastlab" type="lab2.py"/>
  </group>
  
</launch>
